timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -344,6 +239,55 @@ def get_model_names(self, level: Optional[int] = None, **kwargs) -> List[str]:
             return list(node for node, l in self._get_model_levels().items() if l == level)  # noqa: E741
         return list(self.model_graph.nodes)
 
+    def get_models_attribute_dict(self, attribute: str, models: Optional[List[str]] = None) -> Dict[str, Any]:
+        """Get an attribute from the `model_graph` for each of the model names
+        specified. If `models` is none, the attribute will be returned for all models"""
+        results = {}
+        if models is None:
+            models = self.get_model_names()
+        for model in models:
+            results[model] = self.model_graph.nodes[model][attribute]
+        return results
+
+    def get_models_info(self, models: Optional[List[str | AbstractTimeSeriesModel]] = None) -> Dict[str, Any]:
+        models_ = models if models is not None else self.get_model_names()
+        model_info_dict = dict()
+        for model in models_:
+            if isinstance(model, str):
+                if model in self.models.keys():
+                    model = self.models[model]
+                model_type = self.get_model_attribute(model=model, attribute="type")
+                model_path = os.path.join(self.path, self.get_model_attribute(model=model, attribute="path"))
+                model_info_dict[model] = model_type.load_info(path=model_path)
+            else:
+                model_info_dict[model.name] = model.get_info()
+        return model_info_dict
```

Comment:
These two methods could, in theory, be reconciled between `AbstractTabularTrainer` and `AbstractTimeSeriesTrainer`. Do you think that's possible in this PR? Or should we at least add a TODO?

Reply:
I don't think we should try extra hard. In principle the two implementations serve (i.e., are responsible to) different users. The contract is there to 1/ make it easier for users to switch from one module to another and 2/ force us to not drift apart too much in behavior and not do very obvious code duplication.

Reply:
These two methods looked pretty close, but I'm also fine with keeping them as-is.

Reply:
Worked more on it. `get_models_attribute_dict` changes behavior for our `fit_summary()` API method ever so slightly, and quite likely doesn't affect tabular's behavior. Note that we can see this as a fix. Tabular worked inconsistently when `get_models_attribute_dict("path", models=None)` and `get_models_attribute_dict("path", models=models)` was called.